### ISSUE TITLE
UselessVariableSniff overlooks some useless vars

### ISSUE THREAD

**Issue by Majkl578** [slevomat/coding-standard#445]
These were not detected:

```php
        $sql .= 'EXISTS (' . $this->walkSubselect($existsExpr->subselect) . ')';

        return $sql;
```

https://github.com/doctrine/doctrine2/blob/f4c84897baf737933926bbe507726d3f0d949636/lib/Doctrine/ORM/Query/SqlWalker.php#L1936-L1943

```php
        $sql .= ')';

        return $sql;
```

https://github.com/doctrine/doctrine2/blob/f4c84897baf737933926bbe507726d3f0d949636/lib/Doctrine/ORM/Query/SqlWalker.php#L2119-L2130

**Comment by kukulich** [comment]
This is currently feature but if you think it should be reported too I can add option.

**Comment by Majkl578** [comment]
Hmm, where do you draw the line what's feature and what isn't? This one is reported for example: https://github.com/doctrine/doctrine2/blob/f4c84897baf737933926bbe507726d3f0d949636/lib/Doctrine/ORM/Query/SqlWalker.php#L2197-L2200

**Comment by Majkl578** [comment]
I guess it depends on how agressive the sniff should be, technically also some ifs/ternaries could be inlined too. I was just a bit confused to see some more complex statements reported and a single one (the one with `.= ')'`) not.

**Comment by kukulich** [comment]
I will try to not report https://github.com/doctrine/doctrine2/blob/f4c84897baf737933926bbe507726d3f0d949636/lib/Doctrine/ORM/Query/SqlWalker.php#L2197-L2200

**Comment by kukulich** [comment]
Fixed in https://github.com/slevomat/coding-standard/commit/360ed8f9ce03afbf07fc67d198fe187fc2543c32
